packages/mco/components/modals/app-manage-policies/app-manage-policies-modal.tsx

```diff
@@ -3,51 +3,88 @@ import { StatusBox } from '@odf/shared/generic/status-box';
 import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
 import { Modal, ModalVariant } from '@patternfly/react-core';
 import { PolicyListView } from './policy-list-view';
+import { UnAssignPolicyView } from './unassign-policy-view';
 import {
+  ManagePolicyStateAction,
   ManagePolicyStateType,
   MessageType,
   ModalActionContext,
   ModalViewContext,
   initialPolicyState,
   managePolicyStateReducer,
+  ManagePolicyState,
 } from './utils/reducer';
 import { ApplicationType } from './utils/types';
 
-export const AppManagePoliciesModal: React.FC<AppManagePoliciesModalProps> = ({
+export const ModalBody: React.FC<ModalBodyProps> = ({
   applicaitonInfo,
-  loaded,
-  loadError,
-  isOpen,
-  close,
+  state,
+  dispatch,
 }) => {
-  const [state, dispatch] = React.useReducer(
-    managePolicyStateReducer,
-    initialPolicyState
-  );
-  const { t } = useCustomTranslation();
-
   const setModalContext = (modalViewContext: ModalViewContext) => {
     dispatch({
       type: ManagePolicyStateType.SET_MODAL_VIEW_CONTEXT,
       payload: modalViewContext,
     });
   };
 
-  const setModalActionContext = (modalViewContext: ModalActionContext) =>
+  const setModalActionContext = (
+    modalActionViewContext: ModalActionContext,
+    modalViewContext?: ModalViewContext
+  ) =>
     dispatch({
       type: ManagePolicyStateType.SET_MODAL_ACTION_CONTEXT,
-      context: state.modalViewContext,
-      payload: modalViewContext,
+      context: modalViewContext || state.modalViewContext,
+      payload: modalActionViewContext,
     });
 
-  const setMessage = (message: MessageType) => {
+  const setMessage = (
+    message: MessageType,
+    modalViewContext?: ModalViewContext
+  ) => {
     dispatch({
       type: ManagePolicyStateType.SET_MESSAGE,
-      context: state.modalViewContext,
+      context: modalViewContext || state.modalViewContext,
       payload: message,
     });
   };
 
+  return (
+    (state.modalViewContext === ModalViewContext.POLICY_LIST_VIEW && (
+      <PolicyListView
+        dataPolicyInfo={applicaitonInfo?.dataPolicies}
+        state={state.policyListView}
+        dispatch={dispatch}
+        setModalContext={setModalContext}
+        setModalActionContext={setModalActionContext}
+        setMessage={setMessage}
+      />
+    )) ||
+    (state.modalViewContext === ModalViewContext.UNASSIGN_POLICY_VIEW && (
+      <UnAssignPolicyView
+        state={state.unAssignPolicyView}
+        dispatch={dispatch}
+        setModalContext={setModalContext}
+        setModalActionContext={setModalActionContext}
+        setMessage={setMessage}
+      />
+    ))
+  );
```

Comment:
nits (to save OR/AND conditions):
```suggestion  
     {
        {
          [ModalViewContext.POLICY_LIST_VIEW]: (
              <PolicyListView
                dataPolicyInfo={applicaitonInfo?.dataPolicies}
                state={state.policyListView}
                dispatch={dispatch}
                setModalContext={setModalContext}
                setModalActionContext={setModalActionContext}
                setMessage={setMessage}
              />
          ),
          [ModalViewContext.UNASSIGN_POLICY_VIEW ]: (
              <UnAssignPolicyView
                state={state.unAssignPolicyView}
                dispatch={dispatch}
                setModalContext={setModalContext}
                setModalActionContext={setModalActionContext}
                setMessage={setMessage}
              />
          ),
        }[state.modalViewContext]
      }
```

Reply:
switch case in JSX...